va/va_trace.c

```diff
@@ -5842,3 +5866,95 @@ void va_TraceStatus(VADisplay dpy, const char * funcName, VAStatus status)
     va_TraceMsg(trace_ctx, "=========%s ret = %s, %s \n", funcName, vaStatusStr(status), vaErrorStr(status));
     DPY2TRACE_VIRCTX_EXIT(pva_trace);
 }
+
+void va_TraceEvent(
+    VADisplay dpy,
+    unsigned short id,
+    unsigned short opcode,
+    unsigned int num,
+    VAEventData *desc
+)
+{
+    struct va_trace *pva_trace = (struct va_trace *)(((VADisplayContextP)dpy)->vatrace);
+    int data[VA_TRACE_MAX_SIZE / sizeof(int)];
+    size_t write_size = 0;
+    char *p_data;
+    int i;
+
+    if (pva_trace == NULL || pva_trace->ftrace_fd < 0) {
+        return;
+    }
+    data[0] = VA_TRACE_ID;
```

Comment:
ftrace implement, write event data into ftrace entry point directly.
the header is 12 byte to identify libva trace, trace tool rely on this header to identify libva trace